### PR TITLE
feat(cli): `db seed` command

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- implement `db seed` command to:
  - create database if not existed
  - create tables and seed data

### Normal

After `logto db seed`:
- new database created
- tables created
- data seeded

### Error handling

Manually make an error in the midway:

image

Check database, nothing changed

### CODE REVIEW

File: packages/cli/src/commands/database/seed.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      log.error(error);
+    }
+
+    const databaseUrl = await getDatabaseUrlFromConfig();
+    const dsn = parseDsn(databaseUrl);
+    // It's ok to fall back to '?' since:
+    // - Database name is required to connect in the previous pool
+    // - It will throw error when creating database using '?'
+    const databaseName = dsn.databaseName ?? '?';
+    const maintenancePool = await createPool(stringifyDsn({ ...dsn, databaseName: 'postgres' }));
```
Comment (charIeszhao) [inline packages/cli/src/commands/database/seed.ts:44]: Should we allow custom database name, instead of hard-coded `postgres`?

**Review by wangsijie — APPROVED** [review]

File: packages/cli/src/database.ts [hunk 2]
```diff
@@ -37,3 +45,61 @@ export const convertToIdentifiers = <T extends Table>({ table, fields }: T, with
...
+/**
+ * Note `undefined` is removed from the acceptable list,
+ * since you should NOT call this function if ignoring the field is the desired behavior.
+ * Calling this function with `null` means an explicit `null` setting in database is expected.
+ * @param key The key of value. Will treat as `timestamp` if it ends with `_at` or 'At' AND value is a number;
+ * @param value The value to convert.
+ * @returns A primitive that can be saved into database.
+ */
+// eslint-disable-next-line complexity
+export const convertToPrimitiveOrSql = (
```
Comment (darcyYe) [inline packages/cli/src/database.ts:58]: This seems a duplication of `convertToPrimitiveOrSql` in @logto/core, should we move this to some other packages (such as @logto/shared) as a global util function?

File: packages/cli/src/commands/database/seed.ts (hunk 1 above)
Comment (gao-sun) [reply packages/cli/src/commands/database/seed.ts:44]: it's the maintenance database name, in most cases it'll be `postgres`. customization is good, should we implement until ask?

File: packages/cli/src/database.ts (hunk 2 above)
Comment (gao-sun) [reply packages/cli/src/database.ts:58]: image

File: packages/cli/src/commands/database/seed.ts (hunk 1 above)
Comment (charIeszhao) [reply packages/cli/src/commands/database/seed.ts:44]: ok, I'm good with it

**Review by darcyYe — APPROVED** [review]
LGTM

**Pull request merged**